src/App.tsx

```diff
@@ -1,54 +1,115 @@
+
+  const delay = 1000;
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const showDropdown = useCallback(
+    debounce(setIsActive, delay),
+    [],
+  );
+
+  const handleFocus = () => {
+    setIsActive(true);
+  };
+
+  const handleSelected = (person: Person) => {
+    setSelectedPerson(person);
+    setAppliedQuery('');
+    setIsActive(false);
+    setQuery('');
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value);
+    setIsActive(false);
+    applyQuery(e.target.value);
+    showDropdown(true);
+  };
+
+  const visiblePeople = useMemo(() => {
+    return peopleFromServer
+      .filter(person => person.name.toLowerCase()
+        .includes(appliedQuery.toLowerCase()));
+  }, [appliedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+          : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
+      <div
+        className={classNames('dropdown', {
+          'is-active': isActive,
+        })}
+      >
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleChange}
+            onFocus={handleFocus}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {visiblePeople.length
+              ? (visiblePeople.map(person => (
+                <div
+                  className="dropdown-item"
+                  key={person.name}
+                >
+                  <a
+                    href="http//"
+                    className={person.sex === 'm'
+                      ? 'has-text-link'
+                      : 'has-text-danger'}
```

Comment:
Use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally

Reply:
done